### PR TITLE
[2023.09.25.00] Backport patch to avoid corrupted osx_arm64 builds 

### PR DESCRIPTION
**Pull request by jjerphan**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

For context see:
 - https://github.com/man-group/ArcticDB/issues/1322
 - https://github.com/facebook/folly/commit/e3dc03f5621935bf9caabadb9f46712f561747e9
 - https://github.com/llvm/llvm-project/issues/73313

### CODE REVIEW

**Comment by jjerphan** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml
Comment (h-vetinari) [inline recipe/meta.yaml:15]: nit: document the patch provenance in `meta.yaml`
```suggestion
    # backport https://github.com/facebook/folly/commit/e3dc03f5621935bf9caabadb9f46712f561747e9
    - 0001-cut-outdated-apple-version-checks.patch
```

**Review by h-vetinari — APPROVED** [review]
Not a maintainer here (I know that @xhochy is on holiday), but LGTM aside from a trivial nit.

**Comment by jjerphan** [comment]
I more or less have been responsible for this branch for ArcticDB's needs.

I propose to merge this PR when the workflows complete and that I take care of any failures this might cause in the ecosystem (unlikely there will exist, IMO).

**Pull request merged**
